src/runtime_src/tools/xclbin/XclBinUtilities.h

```diff
@@ -40,6 +48,54 @@ std::string format(const std::string& format, Args ... args) {
   return std::string(buf.get(), buf.get() + size);
 }
 
+class XclBinUtilException : public std::runtime_error {
+  private:
+    std::string m_msg;
+    std::string m_file;
+    int m_line;
+    std::string m_function;
+    XclBinExceptionType m_eExceptionType;
+
+public:
+    XclBinUtilException(XclBinExceptionType _eExceptionType, const std::string & _msg, const char * _file, int _line, const char * _function) 
+    : std::runtime_error(_msg)
+    , m_msg(_msg)
+    , m_file(_file)
+    , m_line(_line)
+    , m_function(_function)
+    , m_eExceptionType(_eExceptionType) {
+      // Empty
+    }
+
+    ~XclBinUtilException() throw() 
+    {
+      // Empty
+    }
+
+    const char *msg() const throw() {
+      return m_msg.c_str();
+    }
+
+    const char *file() const throw() {
+      return m_file.c_str();
+    }
+
+    int line() const throw() {
+      return m_line;
+    }
+
+    const char *function() const throw() {
+      return m_function.c_str();
+    }
+
+    XclBinExceptionType exceptionType() const throw() {
+      return m_eExceptionType;
+    }
+};
+
+#define throw_xclbin_exception(exceptionType, msg) throw XclBinUtilities::XclBinUtilException(exceptionType, msg, __FILE__, __LINE__, __FUNCTION__);
```

Comment:
I think code will reader nicer if "throw" is part moved out of the macro and it was called like
"throw xclbin_exception(blah,blah);"

Reply:
The code is nicer and I have dropped the macro.  The code has been refactored to throw the XclBinUtilException class (e.g., throw XUtil::XclBinUtilException(XET_MISSING_SECTION, errMsg);)